### ISSUE TITLE
Hard to debug deploy failures

### ISSUE THREAD

**Issue by gordonbisnor** [boldpoker/edeliver#37]
I'm getting a variety of hard to debug deploy failures. I've had some success but lots of failure, and it has been hard to tell why some things are working and others not. 

- I'm attempting to build and deploy to the same server, and use my local machine as the release store. 
- I tried removing the application, and the builds folder from my production sever, as well as removing all releases from my local .deliver folder

**This is my .edeliver/config:**

```bash
#!/usr/bin/env bash

APP="xxx" # name of your release

BUILD_HOST="xxx" # host where to build the release
BUILD_USER="xxx" # local user at build host
BUILD_AT="/home/xxx/builds" # build directory on build host

PRODUCTION_HOSTS="xxx" # deploy / production hosts separated by space
PRODUCTION_USER="xxx" # local user at deploy hosts
DELIVER_TO="/home/xxx/xxx/" # deploy directory on production hosts

GIT_CLEAN_PATHS=${GIT_CLEAN_PATHS:="rel"}
```

**I then run mix edeliver build release --branch=master --verbose and I get the following error:**

```bash
using mix to generate release
Building release with MIX_ENV=prod.
This is an upgrade, verifying appups exist for updated dependencies..
==> All dependencies have appups ready for release!
==> Generated .appup for xxx 0.0.1 -> 0.0.7

FAILED 1:

    [ -f ~/.profile ] && source ~/.profile
    set -e
    cd /home/xxx/builds
    if [ "mix" = "rebar" ]; then
      echo "using rebar to generate release"
      ./rebar   -f generate
    elif [ "mix" = "relx" ]; then
      echo "using relx to generate release"
      ./relx release
    elif [ "mix" = "mix" ]; then
      echo "using mix to generate release"
      MIX_ENV="prod" LINK_SYS_CONFIG="" LINK_VM_ARGS="" APP="xxx" mix release  --no-confirm-missing
    fi
```

Don't know how to debug based on that that message. I've been getting similar failure messages when trying to do releases and upgrades. I'm sure there must be something wrong with my config file or server setup but hard to know what. 

Any help appreciated. I assume this is not an edeliver error and if so I can post this in StackOverflow if need be.

**Comment by gordonbisnor** [comment]
Additionally, after getting that failure, I logged in to the  server, visited builds/bin and ran ./xxx start and the application started. Wondering if the empty  LINK_SYS_CONFIG="" LINK_VM_ARGS="" could have anything to do with this.

**Comment by gordonbisnor** [comment]
It seems like I have this worked out now. I removed everything related to the app deployment from my server, then tinkered slightly with the config file:

```bash
#!/usr/bin/env bash
APP="xxx" # name of your release
BUILD_HOST="xxx" # host where to build the release
BUILD_USER="xxx" # local user at build host
BUILD_AT="/tmp/erlang/xxx/builds" # build directory on build host
PRODUCTION_HOSTS="xxx" # deploy / production hosts separated by space
PRODUCTION_USER="xxx" # local user at deploy hosts
DELIVER_TO="/home/xxx/apps" # deploy directory on production hosts
```

**Then I did a first release with this:**

```bash
mix edeliver build release
mix edeliver deploy release to production --version=0.0.7
mix edeliver start production
```

**And then subsequently upgraded with this:**

```bash
mix edeliver build upgrade --with=0.0.7 --to=0.0.9
mix edeliver deploy upgrade to production --version=0.0.9
```

**Comment by bharendt** [comment]
True, I think it was hard to debug but __exrm sometimes fails silently__, mostly during the relup generation.

So I __reopen the issue__ and will __add the `--verbosity=verbose` option to__ the exrm __release task__ if edeliver is used in verbose mode.

Actually, if the `GIT_CLEAN_PATHS` is `rel`, nothing from a previous release should exist before building a new release, which means __actually no relup should be created__. But we __will check__ that too, __maybe it depends on the git version__. Maybe you could test on your build host whether `git clean -fdx rel` really removes the folder?

**Comment by bharendt** [comment]
Verified that `git clean -fdx rel` used if `GIT_CLEAN_PATHS="rel"` works as expeced with git versions `2.4.9` and `1.9.1`

**Comment by gordonbisnor** [comment]
I had a seemingly unrelated  [issue](http://stackoverflow.com/questions/34598539/phoenix-error-in-deployment-module-connection-is-not-available/34598838#34598838) where the app was able to run, but certain views controllers were producing errors - I wonder if that could have been causing the silent exrm failure.

**Comment by gordonbisnor** [comment]
At any rate, thanks for the feedback, it's very exciting to see a deployment library like this available for Elixir –– thanks for all of the hard work!

**Comment by bharendt** [comment]
We also ran into that `undefined function: Connection.start_link/2` [postgrex issue](http://stackoverflow.com/questions/34598539/phoenix-error-in-deployment-module-connection-is-not-available/34598838#34598838). I think this affects only the runtime of the release, not the building process. But anyway, there are many reasons why building a release or an upgrade might fail, so I think it is a good idea in any way to show more detailed information if that fails. So thanks for the hint.
